pkg/controllers/disruption/consolidation_test.go

```diff
@@ -3583,4 +3584,241 @@ var _ = Describe("Consolidation", func() {
+			})
+			Expect(found).To(BeTrue())
+			Expect(m.GetGauge().GetValue()).To(BeNumerically("~", numNodes))
+			m, found = FindMetricWithLabelValues("karpenter_disruption_eligible_nodes", map[string]string{
+				"method":             "consolidation",
+				"consolidation_type": "empty",
+			})
+			Expect(found).To(BeTrue())
+			Expect(m.GetGauge().GetValue()).To(BeNumerically("~", numNodes))
+
+		})
+		It("should update metrics for single-nodeclaim consolidation", func() {
+			pods := test.Pods(numNodes, test.PodOptions{
+				ObjectMeta: metav1.ObjectMeta{Labels: labels,
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion:         "apps/v1",
+							Kind:               "ReplicaSet",
+							Name:               rs.Name,
+							UID:                rs.UID,
+							Controller:         ptr.Bool(true),
+							BlockOwnerDeletion: ptr.Bool(true),
+						},
+					}},
+				ResourceRequirements: v1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						// Make the pods more than half of the allocatable so that only one nodeclaim can be done at any time
+						v1.ResourceCPU: resource.MustParse("20"),
+					},
+				},
+			})
+			ExpectApplied(ctx, env.Client, rs, nodePool)
+			for _, nodeClaim := range nodeClaims {
+				ExpectApplied(ctx, env.Client, nodeClaim)
+			}
+			for _, node := range nodes {
+				ExpectApplied(ctx, env.Client, node)
+			}
+			for i, pod := range pods {
+				ExpectApplied(ctx, env.Client, pod)
+				ExpectManualBinding(ctx, env.Client, pod, nodes[i])
+			}
+
+			// inform cluster state about nodes and nodeClaims
+			ExpectMakeNodesAndNodeClaimsInitializedAndStateUpdated(ctx, env.Client, nodeStateController, nodeClaimStateController, nodes, nodeClaims)
+
+			var wg sync.WaitGroup
+			wg.Add(1)
+			finished := atomic.Bool{}
+			go func() {
+				defer GinkgoRecover()
+				defer wg.Done()
+				defer finished.Store(true)
+				ExpectReconcileSucceeded(ctx, disruptionController, client.ObjectKey{})
+			}()
+
+			// advance the clock so that the timeout expires for multi-nodeClaim
+			fakeClock.Step(disruption.MultiNodeConsolidationTimeoutDuration)
+			// advance the clock so that the timeout expires for single-nodeClaim
+			fakeClock.Step(disruption.SingleNodeConsolidationTimeoutDuration)
+
+			ExpectTriggerVerifyAction(&wg)
+
+			// controller should finish
+			Eventually(finished.Load, 10*time.Second).Should(BeTrue())
+			wg.Wait()
+
+			// should have no nodeClaims deleted from single nodeClaim consolidation
+			Expect(ExpectNodeClaims(ctx, env.Client)).To(HaveLen(numNodes))
+
+			m, found := FindMetricWithLabelValues("karpenter_deprovisioning_eligible_machines", map[string]string{
+				"deprovisioner": "consolidation",
+			})
+			Expect(found).To(BeTrue())
+			Expect(m.GetGauge().GetValue()).To(BeNumerically("~", numNodes))
+			m, found = FindMetricWithLabelValues("karpenter_disruption_eligible_nodes", map[string]string{
+				"method":             "consolidation",
+				"consolidation_type": "single",
```

Comment:
For these tests can we refer to the metrics constants that are used rather than using hard-coded strings? Will make sure in the future if these values are changed that these tests are changed appropriately.

Reply:
+1